### PR TITLE
Split side-comments cache into its own collection

### PR DESCRIPTION
**Pull request by jimrandomh**
Split the side-comments cache into its own collection (where previously it was a field on `Posts`). This should improve performance by reducing the (uncompressed) size of what the webserver downloads from the database, since the side-comments cache is large (it contains an annotated copy of the post body) and is only used on post pages, not on the front page (and the front page has a lot of posts displayed on it).

This _doesn't_ drop the old field; that will be in a followup PR, since we want all webservers switched over to the new collection before we drop the old field. Note that the performance benefit comes entirely from dropping the old field, so this won't speed things up at all until that happens.

This includes a manual migration, `populateSideCommentsCache`, which goes through old posts creating a side-comments cache for them. This is necessary because while the cache entries do get recomputed on request, recomputing the cache entry is expensive, and a webcrawler visiting every old post in the archives would consume a stability-threatening amount of CPU time.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204761669261224) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/collections/sideCommentsCache/collection.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export const SideCommentsCache: SideCommentsCachesCollection = createCollection({
+  collectionName: "SideCommentsCaches",
+  typeName: "SideCommentsCache",
+  collectionType: "pg",
+  schema,
+  logChanges: false,
+});
+
+addUniversalFields({collection: SideCommentsCache});
+ensureIndex(SideCommentsCache, { postId: 1, version: 1, generatedAt: 1 });
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/sideCommentsCache/collection.ts:48]: Sounds like we're now going to be storing multiple version of the side comments per post?

File: packages/lesswrong/server/sideComments.ts [hunk 2]
```diff
@@ -4,7 +4,11 @@ import { Comments } from '../lib/collections/comments/collection';
 import type { SideCommentsResolverResult } from '../lib/collections/posts/schema';
 import { getDefaultViewSelector } from '../lib/utils/viewUtils';
 import groupBy from 'lodash/groupBy';
-import some from 'lodash/some';
+import { getToCforPost } from './tableOfContents';
+import { createMutator } from './vulcan-lib/mutators';
+import { SideCommentsCache } from "../lib/collections/sideCommentsCache/collection";
+
+export const sideCommentCacheVersion = 1;
```
Comment (b0b3rt) [inline packages/lesswrong/server/sideComments.ts:11]: This is something like a "format/syntax of side comments" version?

Comment (jimrandomh) [reply packages/lesswrong/server/sideComments.ts:11]: Right. If we bump this version number, all cache entries with the old version number are treated as invalid.

File: packages/lesswrong/lib/collections/sideCommentsCache/collection.ts (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/lib/collections/sideCommentsCache/collection.ts:48]: We check for an existing entry, generate an entry and insert it. So we can get multiple versions iff they fire off at the same time, and have a race condition. I thought about preventing this using an upsert or something, but wasn't sure it was worth the complexity.

**Review by Will-Howard — APPROVED** [review]
Seems good! Sorry for the delay in reviewing, I got confused about who's responsibility it was

**Comment by jimrandomh** [comment]
Looks like I resolved the merge-conflicts in the schema files incorrectly. Reverting, and will try that merge conflict resolution again.

**Pull request merged**